pkg/reconciler/v1alpha1/configuration/configuration_test.go

```diff
@@ -71,209 +65,129 @@ func TestReconcile(t *testing.T) {
+				// Configuration and a Revision is created.
+				WithLatestCreated, WithObservedGen),
 		}},
 		Key: "foo/no-revisions-yet",
 	}, {
 		Name: "webhook validation failure",
 		// If we attempt to create a Revision with a bad ConcurrencyModel set, we fail.
 		WantErr: true,
 		Objects: []runtime.Object{
-			setConcurrencyModel(cfg("validation-failure", "foo", 1234), "Bogus"),
+			cfg("validation-failure", "foo", 1234, WithConfigConcurrencyModel("Bogus")),
 		},
 		WantCreates: []metav1.Object{
-			setRevConcurrencyModel(resources.MakeRevision(cfg("validation-failure", "foo", 1234), nil), "Bogus"),
+			rev("validation-failure", "foo", 1234, WithRevConcurrencyModel("Bogus")),
 		},
 		WantUpdates: []clientgotesting.UpdateActionImpl{{
-			Object: setConcurrencyModel(cfgWithStatus("validation-failure", "foo", 1234,
-				v1alpha1.ConfigurationStatus{
-					Conditions: duckv1alpha1.Conditions{{
-						Type:     v1alpha1.ConfigurationConditionReady,
-						Status:   corev1.ConditionFalse,
-						Reason:   "RevisionFailed",
-						Message:  `Revision creation failed with message: "invalid value \"Bogus\": spec.concurrencyModel".`,
-						Severity: "Error",
-					}},
-				}), "Bogus"),
+			Object: cfg("validation-failure", "foo", 1234, WithConfigConcurrencyModel("Bogus"),
+				// Expect Revision creation to fail with the following error.
+				MarkRevisionCreationFailed(`invalid value "Bogus": spec.concurrencyModel`)),
 		}},
 		Key: "foo/validation-failure",
 	}, {
 		Name: "elide build when a matching one already exists",
 		Objects: []runtime.Object{
-			cfgWithBuild("need-rev-and-build", "foo", 99998, &buildSpec),
+			cfg("need-rev-and-build", "foo", 99998, WithBuild),
 			// An existing build is reused!
-			resources.MakeBuild(cfgWithBuild("something-else", "foo", 12345, &buildSpec)),
+			resources.MakeBuild(cfg("something-else", "foo", 12345, WithBuild)),
 		},
 		WantCreates: []metav1.Object{
-			resources.MakeRevision(cfgWithBuild("need-rev-and-build", "foo", 99998, &buildSpec), &corev1.ObjectReference{
-				APIVersion: "build.knative.dev/v1alpha1",
-				Kind:       "Build",
-				Name:       "something-else-12345",
-			}),
+			rev("need-rev-and-build", "foo", 99998, WithBuildRef("something-else-12345")),
 		},
 		WantUpdates: []clientgotesting.UpdateActionImpl{{
-			Object: cfgWithBuildAndStatus("need-rev-and-build", "foo", 99998, &buildSpec,
-				v1alpha1.ConfigurationStatus{
-					LatestCreatedRevisionName: "need-rev-and-build-99998",
-					ObservedGeneration:        99998,
-					Conditions: duckv1alpha1.Conditions{{
-						Type:     v1alpha1.ConfigurationConditionReady,
-						Status:   corev1.ConditionUnknown,
-						Severity: "Error",
-					}},
-				},
-			),
+			Object: cfg("need-rev-and-build", "foo", 99998, WithBuild,
+				// The following properties are set when we first reconcile a Configuration
+				// that stamps out a Revision with an existing Build.
+				WithLatestCreated, WithObservedGen),
```

Comment:
Same as the above comment on WithLatestCreated.

Reply:
https://github.com/knative/serving/blob/d8283a8db83525430f093a8d2bc5e77e2605685d/pkg/apis/serving/v1alpha1/configuration_types_test.go#L197